src/broker/brokercfg.c

```diff
@@ -286,25 +286,37 @@ static int validate_queues_config (flux_conf_t *conf, flux_error_t *error)
         json_object_foreach (queues, name, entry) {
             json_error_t jerror;
             json_t *policy = NULL;
+            json_t *requires = NULL;
 
             if (json_unpack_ex (entry,
                                 &jerror,
                                 0,
-                                "{s?o !}",
-                                "policy", &policy) < 0) {
+                                "{s?o s?o !}",
+                                "policy", &policy,
+                                "requires", &requires) < 0) {
                 errprintf (error, "queues.%s: %s", name, jerror.text);
-                errno = EINVAL;
-                return -1;
+                goto inval;
             }
             if (policy) {
                 char key[1024];
                 snprintf (key, sizeof (key), "queues.%s.policy", name);
                 if (validate_policy_json (policy, key, error) < 0)
                     return -1;
             }
+            if (requires) {
```

Comment:
One thing we might want to do here is validate that each configured property is valid according to [RFC 20](https://flux-framework.readthedocs.io/projects/flux-rfc/en/latest/spec_20.html), i.e. does not contain whitespace or any characters from the set `! & ' " ^ ` | ( )`.

Reply:
Ah, I was wrong! Actually `^` should be allowed since that is currently our shorthand for `not`, e.g. a sysadmin should be able to configure a `queue.NAME.requires` of `[ "^batch" ]` which means that queue  `NAME` excludes all resources with the `batch` property.

I'm sorry for messing that one up.